lib/yust_service.dart

```diff
@@ -531,8 +561,21 @@ class YustService {
             // If an empty list is passed, an error is thrown.
             // I think that it should behave the same and return no data.
 
-            if (operand2 != null && operand2 is List && operand2.isEmpty) {
-              operand2 = null;
+            if (operand2 == null) {
+            } else if (operand2 is Iterable) {
+              if ((operand2 as Iterable).isEmpty) {
+                operand2 = null;
+              } else {
+                final List typedOperand2 = operand2.toList();
+                final onlyStringElements = typedOperand2.whereType<String>();
+
+                assert(
+                  typedOperand2.length == onlyStringElements.length,
```

Comment:
Prüfst du, ob Operand 2 ein String ist? Der Operand 2 muss doch kein String sein.

Reply:
Wenn der zweite Operand ein Iterable also z.B. eine Liste ist, überprüfe ich, dass alle seine Elemente Strings sind. Wir hatten nämlich häufiger das Problem, dass wir aus Versehen ein Enum in einer Liste als zweiten Operanden übergeben haben, dabei hätte der Enum-Wert vorher noch in einen String übersetzt werden müssen. Dann haben wir einfach keine Ergebnisse bekommen und mussten lange nach dem Problem suchen. Wenn stattdessen hier direkt ein Fehler geworfen wird, wenn ein nicht akzeptabler Typ ankommt, findet man den Fehler fast sofort.

Wenn die Überprüfung auf bloß einen String zu eingeschränt ist, kann man sie ja auf die ganzen von Firebase unterstützen Grunddatentypen erweitern (Boolean, Number, Timestamp usw.). Dann wäre noch immer der schnelle Hinweis auf den Fehler gewährleistet, dass man aus Versehen einen Enum-Wert direkt übergeben hat.